admin/auth_token.go

```diff
@@ -175,20 +175,27 @@ func (t *magicAuthToken) OwnerID() string {
 
 // IssueMagicAuthTokenOptions provides options for IssueMagicAuthToken.
 type IssueMagicAuthTokenOptions struct {
-	ProjectID       string
-	TTL             *time.Duration
-	CreatedByUserID *string
-	Attributes      map[string]any
-	FilterJSON      string
-	Fields          []string
-	State           string
-	DisplayName     string
-	Internal        bool
-	Resources       []database.ResourceName
+	ProjectID                string
+	TTL                      *time.Duration
+	CreatedByUserID          *string
+	Attributes               map[string]any
+	FilterJSON               string
+	Fields                   []string
+	State                    string
+	DisplayName              string
+	Internal                 bool
+	Resources                []database.ResourceName
+	TransitiveAccessResource *database.ResourceName // Optional, used for transitive access
```

Comment:
Could we grant transitive access to all resources in `Resources`, so we don't need a separate field?